libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -547,10 +567,14 @@ def call_model(state: AgentState, config: RunnableConfig) -> AgentState:
 
     async def acall_model(state: AgentState, config: RunnableConfig) -> AgentState:
         response = await model_runnable.ainvoke(state, config)
-        if (
-            state["is_last_step"]
-            and isinstance(response, AIMessage)
-            and response.tool_calls
+        has_tool_calls = isinstance(response, AIMessage) and response.tool_calls
+        all_tools_return_direct = (
+            all(call["name"] in should_return_direct for call in response.tool_calls)
+            if isinstance(response, AIMessage)
+            else False
+        )
+        if (state["remaining_steps"] < 1 and all_tools_return_direct) or (
```

Comment:
shouldn't these be the same between the call_model and acall_model??

Reply:
oops